app/controllers/api_controller.rb

```diff
@@ -94,8 +94,8 @@ def set_headers
     + request.env['PATH_INFO'] + ".json'"
   end
 
-  def authenticator
-    Authentication::UserAuthenticator.new(username: username, password: password_header)
+  def user_authenticator
+    Authentication::UserAuthenticator::Db.new(username: username, password: password_header)
```

Comment:
today it's also possible so authenticate with LDAP credentials to this endpoint. 
if you restrict this to DB only, this is not possible any more. 
I think you should carefully think this through. It's is likely that your implementation works as expected. 
I think it should not be possible for root to login here by header.